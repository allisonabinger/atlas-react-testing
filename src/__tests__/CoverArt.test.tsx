// Test Suite for the CoverArt Component

import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import CoverArt from "../components/CoverArt";
import SoulCanvas from "../assets/SoulCanvas.jpeg";

// Snapshot Tests for CoverArt

test("CoverArt renders an image", () => {
  const { container } = render(<CoverArt imageURL={SoulCanvas} />);
  expect(container).toMatchSnapshot();
});

test("CoverArt will not render without a URL", () => {
  const { container } = render(<CoverArt imageURL="" />);
  expect(container).toMatchSnapshot();
});

test("CoverArt will not render without a valid URL", () => {
  const { container } = render(<CoverArt imageURL="525600" />);
  expect(container).toMatchSnapshot();
});
