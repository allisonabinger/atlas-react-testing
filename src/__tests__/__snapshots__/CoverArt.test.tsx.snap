// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CoverArt renders an image 1`] = `
<div>
  <div
    class="flex cover-art mb-6 w-full justify-center"
  >
    <img
      alt="Current Album Cover"
      class="rounded-md md:w-400 md:h-400"
      src="/src/assets/SoulCanvas.jpeg"
    />
  </div>
</div>
`;

exports[`CoverArt will not render without a URL 1`] = `
<div>
  <div
    class="flex cover-art mb-6 w-full justify-center"
  >
    <img
      alt="Current Album Cover"
      class="rounded-md md:w-400 md:h-400"
      src=""
    />
  </div>
</div>
`;

exports[`CoverArt will not render without a valid URL 1`] = `
<div>
  <div
    class="flex cover-art mb-6 w-full justify-center"
  >
    <img
      alt="Current Album Cover"
      class="rounded-md md:w-400 md:h-400"
      src="525600"
    />
  </div>
</div>
`;
